test/common_nn.py

```diff
@@ -263,7 +263,11 @@ def nlllossNd_reference(input, target, weight=None, ignore_index=-100,
     N = input.size(0)
     C = input.size(1)
     out_size = (N,) + input.size()[2:]
-    output = Variable(torch.zeros(out_size).type_as(input))
+    output = torch.zeros(out_size).type_as(input)
+    if isinstance(target, Variable):
+        target = target.data
```

Comment:
why are these necessary?  I thought .data is basically a no-op.

Reply:
Yeah, it's not necessary here. I was originally trying to separate out some of these changes into separate commits. I'll remove it.